### DISCUSSION TITLE
Inverse PINN Problem - General Questions

### DISCUSSION THREAD

**Discussion by KostasVog20** [lululxvi/deepxde#1644] [Q&A]
Hello everyone, I just wanted to express how incredible the DeepXDE package is! It truly stands out as an exceptional tool.

I am endeavoring to utilize the DeepXDE package for solving an inverse problem involving parameter identification. Specifically, I am dealing with a system of three coupled partial differential equations (PDEs) where two parameters, Deff and um, need to be identified. The general form of my system is:

T = y[:, 0:1]                                                                                           # temperature                                 
S1 = y[:, 1:2]                                                                                         # endothermic fuel
S2 = y[:, 2:3]                                                                                         # exothermic fuel                                
dT_x = dde.grad.jacobian(y, x, i = 0, j = 0)                                           # dT/dx
dT_xx = dde.grad.hessian(y, x, component = 0, i = 0, j = 0)                # d^2T/dx^2
dT_t = dde.grad.jacobian(y, x, i = 0, j = 1)                                            # dT/dt
dS1_t = dde.grad.jacobian(y, x, i = 1, j = 1)                                          # dS1/dt
dS2_t = dde.grad.jacobian(y, x, i = 2, j = 1)                                          # dS2/dt

Eq1 = (dT_t - 0.38 * (Deff * dT_xx - um * dT_x) + 938.46 * S1 * 30 * tf.exp( - 4500 / T) - 8546.15 * S2 * ((40 * tf.exp( - 7000 / T) * 0.002) / (40 * tf.exp( - 7000 / T) + 0.002)) + 0.009 * (T - 300))
Eq2 = dS1_t + S1 * 30 * tf.exp( - 4500 / T)                                        
Eq3 = dS2_t + S2 * ((40 * tf.exp(- 7000 / T) * 0.002)/(40 * tf.exp( - 7000 / T) + 0.002))

This is an 1D spatiotemporal problem with **x in [0, 100] m** and **t in [0, 200] s**. The temperature values are in the interval of [300, 2000] K and S1, S2 in [0, 1]. To train the model, I generated data like: x = [0, 1, 2, .........., 100] and t = [0, 1, 2, ......., 200] and T, S1, S2 are the corresponding matrices with values at these specific spatial and temporal points. I have several general questions that I'd like to ask, and I would appreciate it if you could provide answers:

1) The primary question revolves around whether it's necessary to rescale the problem to fit a geometry of [0, 1] and a time interval of [0, 1]. If rescaling is required, how can this be accomplished? Alternatively, is it possible to retain the current spatial and temporal dimensions and focus on fine-tuning the hyperparameters instead?
2) I'm uncertain about determining suitable values for the parameters num_domain, num_boundary, and num_initial. Additionally, I've utilized "anchors" to sample my specified training points.
3) In reference to the example provided (https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/Navier_Stokes_inverse.py), I've selected data from a smaller subdomain, both spatially and temporally, compared to the original dataset. In this scenario, should I define boundary and initial conditions for my problem? My concern stems from the fact that the training data constitute only a subset of the total dataset and do not encompass points corresponding to boundaries and initial time instances.
4) Is it feasible to apply the "tanh" activation function to all hidden layers and utilize a "relu" activation function for the output layer without necessitating alterations to the original code?
5) I'm having trouble grasping the significance of the "period" variable. Could you provide clarification on its meaning? For instance, if I were to train for 80,000 iterations, what would be an approximate value for this variable?

Your assistance would be greatly appreciated.

**Comment by KostasVog20** [comment]
@lululxvi can you please help me !!

**Comment by lululxvi** [comment]
>     1. The primary question revolves around whether it's necessary to rescale the problem to fit a geometry of [0, 1] and a time interval of [0, 1]. If rescaling is required, how can this be accomplished? Alternatively, is it possible to retain the current spatial and temporal dimensions and focus on fine-tuning the hyperparameters instead?

Not necessary, but usually helps.

>     2. I'm uncertain about determining suitable values for the parameters num_domain, num_boundary, and num_initial. Additionally, I've utilized "anchors" to sample my specified training points.

"anchors" is OK.

>     3. In reference to the example provided (https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/Navier_Stokes_inverse.py), I've selected data from a smaller subdomain, both spatially and temporally, compared to the original dataset. In this scenario, should I define boundary and initial conditions for my problem? My concern stems from the fact that the training data constitute only a subset of the total dataset and do not encompass points corresponding to boundaries and initial time instances.

You can use all training data.

>     4. Is it feasible to apply the "tanh" activation function to all hidden layers and utilize a "relu" activation function for the output layer without necessitating alterations to the original code?

No.

>     5. I'm having trouble grasping the significance of the "period" variable. Could you provide clarification on its meaning? For instance, if I were to train for 80,000 iterations, what would be an approximate value for this variable?

It is the period to print the results.
